src/pages/lessons/[slug].tsx

```diff
@@ -335,17 +337,30 @@ const Lesson: React.FC<LessonProps> = ({initialLesson}) => {
       case 'loaded':
         const viewLimitNotReached = watchCount < MAX_FREE_VIEWS
 
-        if (isEmpty(viewer) && free_forever) {
-          if (viewLimitNotReached && mediaPresent) {
+        // TODO: Detangle this nested series of `if` statements to make the
+        // logic more immediately easy to reason about.
+
+        if (session_id) {
+          // If the URL contains the session ID, even if there is a viewer, put
+          // them in the `subscribing` state.
+          send('SUBSCRIBE')
```

Comment:
This initial check allows a signed-in user who completes checkout to be shown the Thank You overlay instead of immediately loading the lesson for them.